### PR TITLE
Fixed cppChecker reported issues

### PR DESCRIPTION
**Pull request by SinghRajenM**
1. Removed unused variable/code
	- smarthighlighter.cpp, Babtgrid.cpp, shortcut.h, ImageListSet.h

2. Member variable not initialised in ctor
	- ShortcutMapper.h, FileBrowser.h, ProjectPanel.h, TaskList.h, VerticalFileSwitcher.h, WindowDlg.h, lastRecentFileList.h

### CODE REVIEW

File: PowerEditor/src/lastRecentFileList.h [hunk 1]
```diff
@@ -44,6 +44,8 @@ class LastRecentFileList
 public:
 	LastRecentFileList() {
 		_userMax = (NppParameters::getInstance())->getNbMaxRecentFile();
+		for (int i = 0; i < NB_MAX_LRF_FILE; i++)
+			_idFreeArray[i] = false;
 	};
```
Comment (donho) [inline PowerEditor/src/lastRecentFileList.h]: Rather initilaze on its declaration:
bool _idFreeArray[NB_MAX_LRF_FILE] = { false };
https://stackoverflow.com/questions/1920430/c-array-initialization/1920481

Comment (SinghRajenM) [inline PowerEditor/src/lastRecentFileList.h]: I've changed it. But I wonder if it can be complied with VS 2013.

Comment (SinghRajenM) [inline PowerEditor/src/lastRecentFileList.h]: Yes, appveyor build is failing. So reverting the changes.

>c:\projects\notepad-plus-plus\powereditor\src\lastRecentFileList.h(108): error C2536: 'LastRecentFileList::LastRecentFileList::_idFreeArray' : cannot specify explicit initializer for arrays (..\src\ScitillaComponent\Buffer.cpp) [C:\projects\notepad-plus-plus\PowerEditor\visual.net\notepadPlus.vcxproj]
          c:\projects\notepad-plus-plus\powereditor\src\lastRecentFileList.h(108) : see declaration of 'LastRecentFileList::_idFreeArray'
          c:\projects\notepad-plus-plus\powereditor\src\lastRecentFileList.h(108) : see declaration of 'LastRecentFileList::_idFreeArray'

Comment (donho) [inline PowerEditor/src/lastRecentFileList.h]: OK, thank you for your info.
The future release will be compiled by VS2015, as well in appveyor.

**Pull request closed without merging**